src/pages/ArticleList.jsx

```diff
@@ -16,21 +18,78 @@ const ArticleList = () => {
       query.set('topic', slug);
     }
 
-    getAllArticles(query).then(({ articles }) => {
+    const combinedQuery = new URLSearchParams({
+      ...Object.fromEntries(query),
+      ...Object.fromEntries(searchParams),
+    });
+
+    getAllArticles(combinedQuery).then(({ articles }) => {
```

Comment:
I think your api functions were flexible enough to not need changing at this point so bravo 🙌

Reply:
Thanks, I had to think carefully about how I was going to merge the query params in order to maintain use of the  /topics/slug route.